lib/rdoc/code_object/any_method.rb

```diff
@@ -360,15 +360,25 @@ def deduplicate_call_seq(call_seq)
       ignore << is_alias_for.name
       ignore.concat is_alias_for.aliases.map(&:name)
     end
-    ignore.map! { |n| n =~ /\A\[/ ? /\[.*\]/ : n}
+
+    bracket_methods = ignore.grep(/\A\[/)
+    ignore -= bracket_methods
```

Comment:
How about prefixing `ignore_` to `bracket_methods` for readability?

```suggestion
    ignore_bracket_methods, ignore_other_methods = ignore.partition {|i| i.start_with?("[")}
```